### PR TITLE
Add support for sandboxed end-to-end tests with liveview

### PR DESCRIPTION
**Pull request by LostKobrakai**
I've updated `Phoenix.Ecto.SQL.Sandbox` to support usage with LiveView. There are no tests yet however as I'm not really sure how to setup tests with all the `socket/3` magic of `Phoenix.Endpoint`.

### CODE REVIEW

File: lib/phoenix_ecto/sql/sandbox.ex [hunk 1]
```diff
@@ -133,12 +169,39 @@ defmodule Phoenix.Ecto.SQL.Sandbox do
...
+                Enum.find_value(info.x_headers, fn
+                  {^header, val} -> val
+                  _ -> false
+                end)
+            end
+            |> decode_metadata()
+          else
+            %{}
+          end
+        end)
```
Comment (josevalim) [inline lib/phoenix_ecto/sql/sandbox.ex:201]: Great PR, thank you!

However, I would prefer if did not need this code in here. Could we expose something `allow(metadata, sandbox)`? And allow will either:

1. receive a metadata map, which it allows
2. receive a binary which we decode as metadata

This way we can move the whole assign_new and friends to the LiveView itself and not have to worry about LiveView in here. WDYT? The other benefit is that people would be able to use it from channels too.

Comment (LostKobrakai) [reply lib/phoenix_ecto/sql/sandbox.ex:201]: I guess that would work as well. I'd still document an example of the above however.

Comment (josevalim) [reply lib/phoenix_ecto/sql/sandbox.ex:201]: Yes, the documentation is lovely and we should keep it!

**Comment by LostKobrakai** [comment]
I just updated the PR based on the discussion. I hope the code in the docs is correct.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**